src/main/java/pl/javastart/task/FileCalculation.java

```diff
@@ -0,0 +1,78 @@
+package pl.javastart.task;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.List;
+import java.util.Objects;
+
+public class FileCalculation {
+
+    static BigDecimal sumProductsPrice(List<Currency> currencies, List<Product> products) {
+        BigDecimal sum = BigDecimal.ZERO;
+        for (Product product : products) {
+            Currency currency = getCurrency(currencies, product);
+            BigDecimal euroPrice = calculatePriceToEuro(product, Objects.requireNonNull(currency));
+            sum = sum.add(euroPrice);
+        }
+        return sum;
+    }
+
+    static BigDecimal averageValueOfAllProductsInEuro(List<Currency> currencies, List<Product> products) {
+        BigDecimal sum = BigDecimal.ZERO;
+        BigDecimal numberOfProducts = new BigDecimal(products.size());
+        for (Product product : products) {
+            Currency currency = getCurrency(currencies, product);
+            BigDecimal euroPrice = calculatePriceToEuro(product, Objects.requireNonNull(currency));
+            sum = sum.add(euroPrice);
+        }
+        return sum.divide(numberOfProducts, RoundingMode.DOWN);
+    }
+
+    public static BigDecimal theMostExpensiveProductInEuro(List<Product> products, List<Currency> currencies) {
+        BigDecimal highestValue = BigDecimal.valueOf(0);
+        for (Product product : products) {
+            Currency currency = getCurrency(currencies, product);
+            BigDecimal euroPrice = calculatePriceToEuro(product, Objects.requireNonNull(currency));
+            if (euroPrice.compareTo(highestValue) > 0) {
+                highestValue = euroPrice;
+            }
+        }
+        return highestValue;
+    }
+
+    public static BigDecimal theCheapestProductInEuro(List<Product> products, List<Currency> currencies) {
+        BigDecimal cheapestProduct = firstProductValueCalculatedToEuro(currencies, products);
```

Comment:
Po pierwsze, to jeżeli interesuje Cię cena pierwszego produktu, to wydaje się niepotrzebne przekazywać do metody całą listę produktów.
Ten wiersz można by zamienić na:
```
Product firstProduct = products.get(0);
Currency firstProductCurrency = getCurrency(currencies, firstProduct);
BigDecimal cheapestProduct = calculatePriceToEuro(firstProduct, firstProductCurrency);
```
i można wtedy usunąć metodę `firstProductValueCalculatedToEuro`